trino/trino.go

```diff
@@ -976,11 +1022,34 @@ func (qr *driverRows) fetch(allowEOF bool) error {
 	qr.rowindex = 0
 	qr.data = qresp.Data
 	qr.nextURI = qresp.NextURI
+
+	qrStats := QueryProgressCallbackResponse{
+		ID:         qresp.ID,
+		QueryStats: qresp.Stats,
+		Error:      qresp.Error,
+	}
+
 	if len(qr.data) == 0 {
 		if qr.nextURI != "" {
+			if qr.stmt.conn.callback != nil {
+				if qrStats.QueryStats.State == "RUNNING" {
```

Comment:
Why only `RUNNING`?

Reply:
I thought it would be meaningless to make further request on other states since
1. we already make initial progress callback call in beginning
2. make a call when query is finished.

So, at minimum state updated is (QUEUED, FINISHED)

Reply:
The decision on how to process the state should be left to the progress callback.

Reply:
I left comment below as it seems to be similar topic
- https://github.com/trinodb/trino-go-client/pull/14#discussion_r937292355